lua/ttt2/libraries/migrations.lua

```diff
@@ -0,0 +1,62 @@
+---
+-- A TTT2 version migrations library
+-- @author ZenBre4ker
+-- @module migrations
+
+if SERVER then
+	AddCSLuaFile()
+end
+
+migrations = {}
+migrations.folderPath = "terrortown/migrations/"
+migrations.databaseName = "ttt2_migrations"
+migrations.savingKeys = {
+	executedAt = {
+		typ = "string",
+		default = ""
+	}
+}
+
+sql.CreateSqlTable(migrations.databaseName, migrations.savingKeys)
+migrations.orm = orm.Make(migrations.databaseName)
+
+---
+-- Runs all missing forward gamemode migrations.
+-- @realm shared
+function migrations.Apply()
+	local files = file.Find(migrations.folderPath .. "*.lua", "LUA", "nameasc") or {}
+
+	local fileExecutionCounter = 0
+	for i = 1, #files do
+		local fileName = files[i]
+		local fullPath = migrations.folderPath .. files[i]
+
+		local fileInfo = migrations.orm:Find(fileName)
+
+		if fileInfo then continue end
+
+		Dev(1, "Migrating: ", fileName)
+
+		if SERVER then
+			AddCSLuaFile(fullPath)
+		end
+		local isSuccess, errorMessage = pcall(include(fullPath))
+
+		if not isSuccess then
+			error("[TTT2] Migration failed.\n" .. errorMessage, 1)
```

Comment:
```suggestion
			ErrorNoHalt("[TTT2] Migration failed.\n" .. errorMessage, 1)
```

Reply:
This is on purpose to crash the gamemode. `Error` doesnt work, as it is` ErrorNoHalt`. So i chose `error`

Reply:
Ahh, sorry, then nevermind :D